src/Rings/PBWAlgebra.jl

```diff
@@ -20,7 +20,7 @@ mutable struct PBWAlgElem{T, S} <: NCRingElem
   sdata::Singular.spluralg{S}
 end
 
-mutable struct PBWAlgIdeal{D, T, S}
+mutable struct PBWAlgIdeal{D, T, S} <: Ideal{PBWAlgIdeal{T, S}}
```

Comment:
Actually it should be
```suggestion
mutable struct PBWAlgIdeal{D, T, S} <: Ideal{PBWAlgElem{T, S}}
```